app/controllers/events/view.js

```diff
@@ -38,6 +38,20 @@ export default Controller.extend({
           this.notify.error(this.l10n.t('An unexpected error has occurred.'));
         });
       this.set('isEventDeleteModalOpen', false);
+    },
+    copyEvent() {
+      this.set('isCopying', true);
+      this.get('loader')
+        .post(`events/${this.get('model.id')}/copy`, {})
+        .then(copiedEvent => {
+          this.set('isCopying', false);
+          this.transitionToRoute('events.view.edit', copiedEvent.identifier);
+          this.get('notify').success(this.l10n.t('Event copied successfully'));
+        })
+        .catch(() => {
+          this.set('isCopying', false);
```

Comment:
@niranjan94 But since the action will be async, if  we take that 'isCopying' out of the then and catch to use for them both, it would be set to false even if we did not receive response untill that time.